src/synthesizer/emission_models/base_model.py

```diff
@@ -1434,6 +1438,62 @@ def fix_parameters(self, **kwargs):
         """
         self.fixed_parameters.update(kwargs)
 
+    def to_hdf5(self, group):
+        """
+        Save the model to an HDF5 group.
+
+        Args:
+            group (h5py.Group):
+                The group to save the model to.
+        """
+        # First off call the operation to save operation specific attributes
+        # to the group
+        if self._is_extracting:
```

Comment:
I'm not sure why each of these models has to have a dedicated to_hdf5 method, can't an extraction / combination / etc. just have a general `to_hdf5` method, and then call it once here?

Reply:
It reduces the boilerplate to have a targeted writer for each operation. Each operation only has certain attributes populated. If you didn't isolate these then you'd have to have a check for each attr you're writing.

Reply:
Misunderstood, now fixed to use a single private `_to_hdf5` method on each operation